ComposeGallery/src/androidTest/java/com/microsoft/device/display/samples/composegallery/PaneSynchronizationTest.kt

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.device.display.samples.composegallery
+
+import androidx.annotation.StringRes
+import androidx.compose.ui.test.ExperimentalTestApi
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.hasAnyAncestor
+import androidx.compose.ui.test.hasAnySibling
+import androidx.compose.ui.test.hasContentDescription
+import androidx.compose.ui.test.hasScrollAction
+import androidx.compose.ui.test.hasTestTag
+import androidx.compose.ui.test.hasText
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithContentDescription
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollToIndex
+import androidx.lifecycle.ViewModelProvider
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import androidx.window.layout.WindowInfoRepository.Companion.windowInfoRepository
+import androidx.window.testing.layout.WindowLayoutInfoPublisherRule
+import com.microsoft.device.display.samples.composegallery.models.AppStateViewModel
+import com.microsoft.device.display.samples.composegallery.models.DataProvider
+import com.microsoft.device.display.samples.composegallery.ui.ComposeGalleryTheme
+import com.microsoft.device.display.samples.composegallery.ui.view.ComposeGalleryApp
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.rules.TestRule
+
+private const val USE_SWIPE_GESTURE = false
+
+class PaneSynchronizationTest {
+    private val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    private val composeTestRule = createAndroidComposeRule<MainActivity>()
+    private val publisherRule = WindowLayoutInfoPublisherRule()
+
+    @get: Rule
+    val testRule: TestRule
+
+    init {
+        testRule = RuleChain.outerRule(publisherRule).around(composeTestRule)
+        RuleChain.outerRule(composeTestRule)
+    }
+
+    /**
+     * Test that clicking an item in the list pane updates the image shown in the detail pane
+     */
+    @ExperimentalTestApi
+    @Test
+    fun app_testListItemClickUpdatesDetailPane() {
+        val viewModel =
+            ViewModelProvider(composeTestRule.activity).get(AppStateViewModel::class.java)
+        val windowLayoutInfo = composeTestRule.activity.windowInfoRepository().windowLayoutInfo
+
+        composeTestRule.setContent {
+            ComposeGalleryTheme {
+                ComposeGalleryApp(viewModel = viewModel, windowLayoutInfo = windowLayoutInfo)
+            }
+        }
+
+        // Span app so two panes are visible
+        if (USE_SWIPE_GESTURE)
```

Comment:
Instead of hardcoding `USE_SWIPE_GESTURE`, can we have a device check in the `Helper` based on the screen size? If the screen size is recognized, then we do the swipe gesture, if not, we skip the swipe.